adr/0008-store-key-sanitization.md

```diff
@@ -0,0 +1,89 @@
+# 8: Store Key Sanitization
+
+Date: 2024-07-02
+
+## Status
+
+Under Review
+
+## Context
+
+Issues were discovered when accepting arbitrary Keycloak client IDs and setting them into the Store without sanitization. For instance, IDs like `https://google.com/` failed during Store.setItem due to the `/` character, resulting in a `422` error.
+
+
+## Decision
+
+Implementing base64 would be a heavy shift for an extremely important part of the system. Instead, we will sanitize the key by replacing `/` with a character that does not break the `patch` operation before placing the key. We will also increase the tests to find other edge cases before releasing the change.
+
+#### Sanitize using String Replacement
+* Sanitize the key by replacing `/` with a character that does not break the `patch` operation before placing the key.
+* Increasing the tests to find other edge cases before releasing the change.
```

Comment:
It might be nice to think through / add some examples of the sorts of keys we'd like the Store to be able to handle in order to give "other edge cases" (and their respective, future tests) some shape.  Like:
- URIs (cuz that's what drove this PR to begin with)
- dot-delimited strings -- e.g. "a.b.c.d" -- just to be sure our escaping can handle things that look like JSON
- " " (empty strings)
- "null" (JS keywords)
- ...and so on.

Reply:
I hear you on URIs but Store.*Item only accepts strings. 

It can already handle dot-delimited and empty strings. 

The only error we know of until doing more testing and finding more would be url as a string.

```yaml
    hello-pepr-: b #this is empty 
    hello-pepr-a.a.a: a
```